tests/modules/test_text_processor.py

```diff
@@ -260,6 +274,139 @@ def test_subscribe_twice_hindi(self, texting_mock, logging_info_mock, logging_er
     @patch("logging.error")
     @patch("logging.info")
     @patch("modules.text_processor.Texter.send")
+    def test_subscribe_twice_doesnt_change_preg_update_english(self, texting_mock, logging_info_mock, logging_error_mock):
+        t = TextProcessor(phone_number="1-111-1114")
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114").exists())
+        first_response = t.process("JOIN ROSE 25-11-2012")
+        self.assertEqual(first_response, msg_subscribe("English").format(name="Rose"))
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1114")
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114", preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1114")
+        second_response = t2.process("JOIN ROSE 25-11-2012")
+        self.assertTrue(t2.get_contacts().exists())
+        self.assertEqual(second_response, msg_already_sub("English"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1114")
+        logging_error_mock.assert_called_with("Contact for Rose at 1-111-1114 was subscribed but already exists!")
+        contacts = Contact.objects.filter(name="Rose", phone_number="1-111-1114", preg_update=False)
+        self.assertEqual(contacts.count(), 1)
+        self.assertTrue(t2.get_contacts().exists())
```

Comment:
This test doesn't actually seem to test anything about the preg update.

Reply:
Yes it does. It tests whether a second signup text with "JOIN" modifies preg_update in the way that a second text with "BORN" does. Lines 283-4 establish that the contact exists and begins with preg_update False, and Line 291-2 check that after a text with "JOIN" the contact still has preg_update False. If, for example, any new text message from an existing contact updated preg_update to True, this test would not work.

Reply:
Ah gotcha. I was confused. I think something like this (as I've said before)

```
self.assertTrue(t2.get_contacts().first().preg_update)
self.assertTrue(t2.get_contacts().first().preg_signup)
```

could be more explicit than searching.